projects/packages/scheduled-updates/src/class-scheduled-updates.php

```diff
@@ -339,6 +352,67 @@ public static function get_file_mod_capabilities() {
 		return $file_mod_capabilities;
 	}
 
+	/**
+	 * Hook run when a plugin is deleted.
+	 *
+	 * @param string $plugin_file Path to the plugin file relative to the plugins directory.
+	 * @param bool   $deleted     Whether the plugin deletion was successful.
+	 */
+	public static function deleted_plugin( $plugin_file, $deleted ) {
+		require_once ABSPATH . 'wp-admin/includes/update.php';
+
+		if ( ! $deleted ) {
+			return;
+		}
+
+		$events = wp_get_scheduled_events( self::PLUGIN_CRON_HOOK );
+
+		if ( ! count( $events ) ) {
+			return;
+		}
+
+		foreach ( $events as $id => $event ) {
+			// Continue if the plugin is not part of the schedule.
+			if ( ! in_array( $plugin_file, $event->args, true ) ) {
+				continue;
+			}
+
+			// Remove the schedule.
+			$result = wp_unschedule_event( $event->timestamp, self::PLUGIN_CRON_HOOK, $event->args, true );
+
+			if ( is_wp_error( $result ) || false === $result ) {
+				return;
+			}
+
+			$index = array_search( $plugin_file, $event->args, true );
+
+			if ( false !== $index ) {
+				unset( $event->args[ $index ] );
+			}
+
+			if ( count( $event->args ) ) {
```

Comment:
Can we check if `$event->args` is empty and return if so? It'll avoid another level of indentation here